src/internal/streamstore/streamstore.go

```diff
@@ -0,0 +1,198 @@
+// streamstore implements helpers to store large
+// data streams in a repository
+package streamstore
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/stats"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+type streamStore struct {
+	kw      *kopia.Wrapper
+	tenant  string
+	service path.ServiceType
+}
+
+func New(
+	kw *kopia.Wrapper,
+	tenant string,
+	service path.ServiceType,
+) *streamStore {
+	return &streamStore{kw: kw, tenant: tenant, service: service}
+}
+
+const (
+	// detailsItemName is the name of the stream used to store
+	// backup details
+	detailsItemName = "details"
+	// collectionPurposeDetails is used to indicate
+	// what the collection is being used for
+	collectionPurposeDetails = "details"
+)
+
+// WriteBackupDetails persists a `details.Details`
+// object in the stream store
+func (ss *streamStore) WriteBackupDetails(
+	ctx context.Context,
+	backupDetails *details.Details,
+) (string, error) {
+	// construct the path of the container for the `details` item
+	p, err := path.Builder{}.
+		ToServiceCategoryMetadataPath(
+			ss.tenant,
+			collectionPurposeDetails,
+			ss.service,
+			path.DetailsCategory,
+			false,
+		)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: We could use an io.Pipe here to avoid a double copy but that
+	// makes error handling a bit complicated
+	dbytes, err := json.Marshal(backupDetails)
+	if err != nil {
+		return "", errors.Wrap(err, "marshalling backup details")
+	}
+
+	dc := &streamCollection{
+		folderPath: p,
+		item: &streamItem{
+			name: detailsItemName,
+			data: dbytes,
+		},
+	}
+
+	backupStats, _, err := ss.kw.BackupCollections(ctx, nil, []data.Collection{dc}, ss.service)
+	if err != nil {
+		return "", nil
+	}
+
+	return backupStats.SnapshotID, nil
+}
+
+// ReadBackupDetails reads the specified details object
+// from the kopia repository
+func (ss *streamStore) ReadBackupDetails(
+	ctx context.Context,
+	detailsID string,
+) (*details.Details, error) {
+	// construct the path for the `details` item
+	detailsPath, err := path.Builder{}.
+		Append(detailsItemName).
+		ToServiceCategoryMetadataPath(
+			ss.tenant,
+			collectionPurposeDetails,
+			ss.service,
+			path.DetailsCategory,
+			true,
+		)
+	if err != nil {
+		return nil, err
+	}
+
+	var bc stats.ByteCounter
+
+	dcs, err := ss.kw.RestoreMultipleItems(ctx, detailsID, []path.Path{detailsPath}, &bc)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving backup details data")
+	}
+
+	// Expect only 1 data collection
+	if len(dcs) != 1 {
+		return nil, errors.Errorf("expected 1 details data collection: %d", len(dcs))
+	}
+
+	dc := dcs[0]
+
+	select {
```

Comment:
sorry to go back and forth on this, but I think there actually should be a for loop here. It's safe to ignore all the other files that we don't need right now, but for resource management we should drain the `Items()` channel before exiting

Reply:
Easy enough to add back but won't resources get cleaned up after we exit?

I do see a potential correctness issue if we ever decide to write more items in the collection and don't read all of them - the producer could potentially end up blocking. Is that what you're worried about?

Reply:
it's more memory/go routine management. The collections often use another goroutine to send items on the channel. Since there's no `Close()` function and no returned `closer` to call later, the assumption is that all items in the stream will be read. If we don't read all of them, then the goroutine and channel will stick around until Corso exits. Not the end of the world, but will consume some resources

Reply:
Addressed